pkg/deployments/src/task.ts

```diff
@@ -286,14 +286,39 @@ export default class Task {
   }
 
   save(output: RawOutput): void {
+    const parsedOutput = this._parseRawOutput(output);
+
+    if (this.mode === TaskMode.CHECK) {
+      // `save` is only called when by `deploy` (which only happens in LIVE mode), or manually for contracts that are
+      // deployed by other contracts (e.g. Batch Relayer Entrypoints). Therefore, by testing for CHECK mode we can
+      // indentify this second type of contracts, and check them by comparing the saved address to the address that the
+      // task would attempt to save.
```

Comment:
```suggestion
      // `save` is only called by `deploy` (which only happens in LIVE mode), or manually for contracts that are
      // deployed by other contracts (e.g. Batch Relayer Entrypoints). Therefore by testing for CHECK mode we can
      // identify this second type of contracts, and check them by comparing the saved address to the address that the
      // task would attempt to save.
```

This should also get moved up into the dispatcher function described above.